redisvl/extensions/session_manager/semantic_session.py

```diff
@@ -0,0 +1,364 @@
+                {"name": "user_tag", "type": "tag"},
+                {
+                    "name": "vector_field",
+                    "type": "vector",
+                    "attrs": {
+                        "dims": self._vectorizer.dims,
+                        "datatype": "float32",
+                        "distance_metric": "cosine",
+                        "algorithm": "flat",
+                    },
+                },
+            ]
+        )
+
+        self._index = SearchIndex(schema=schema)
+
+        if redis_client:
+            self._index.set_client(redis_client)
+            self._client = redis_client
+        else:
+            self._index.connect(redis_url="redis://localhost:6379")
+            self._client = Redis(decode_responses=True)
+
+        self._index.create(overwrite=False)
+
+        self.set_scope(session_tag, user_tag)
+
+    def set_scope(
+        self,
+        session_tag: Optional[str] = None,
+        user_tag: Optional[str] = None,
+    ) -> None:
+        """Set the tag filter to apply to querries based on the desired scope.
+
+        This new scope persists until another call to set_scope is made, or if
+        scope specified in calls to get_recent or get_relevant.
+
+        Args:
+            session_tag (str): Id of the specific session to filter to. Default is
+                None, which means all sessions will be in scope.
+            user_tag (str): Id of the specific user to filter to. Default is None,
+                which means all users will be in scope.
+        """
+        if not (session_tag or user_tag):
+            return
+        self._session_tag = session_tag or self._session_tag
+        self._user_tag = user_tag or self._user_tag
+        tag_filter = Tag(self.user_field_name) == []
+        if user_tag:
+            tag_filter = tag_filter & (Tag(self.user_field_name) == self._user_tag)
+        if session_tag:
+            tag_filter = tag_filter & (
+                Tag(self.session_field_name) == self._session_tag
+            )
+
+        self._tag_filter = tag_filter
+
+    def clear(self) -> None:
+        """Clears the chat session history."""
+        with self._index.client.pipeline(transaction=False) as pipe:  # type: ignore
+            for key in self._index.client.scan_iter(match=f"{self._index.prefix}:*"):  # type: ignore
+                pipe.delete(key)
+            pipe.execute()
+
+    def delete(self) -> None:
+        """Clear all conversation keys and remove the search index."""
+        self._index.delete(drop=True)
+
+    def drop(self, id_field: Optional[str] = None) -> None:
+        """Remove a specific exchange from the conversation history.
+
+        Args:
+            id_field Optional[str]: The id_field of the entry to delete.
+                If None then the last entry is deleted.
+        """
+        if id_field:
+            key = ":".join([self._index.schema.index.name, id_field])
```

Comment:
`key_separator` is an index schema property that should be used here. The default is `:` but it could be other things. Safest to use the schema's value here as a best practice.